tap_salesforce/salesforce/bulk.py

```diff
@@ -357,3 +354,48 @@ def _iter_lines(self, response):
 
         if pending is not None:
             yield pending
+
+    def _bulk_with_window(self, status_list, catalog_entry, start_date_str, end_date=None, retries=MAX_RETRIES):
+        """Bulk api call with date windowing"""
+        sync_start = singer_utils.now()
+        if end_date is None:
+            end_date = sync_start
+            LOGGER.info("Retrying Bulk Query with PK Chunking")
+        else:
+            LOGGER.info("Retrying Bulk Query with window of date {} to {}".format(start_date_str, end_date.strftime('%Y-%m-%dT%H:%M:%SZ')))
+
+        if retries == 0:
+            raise TapSalesforceException("Ran out of retries attempting to query Salesforce Object {}".format(catalog_entry['stream']))
+
+        job_id = self._create_job(catalog_entry, True)
+        self._add_batch(catalog_entry, job_id, start_date_str, end_date.strftime('%Y-%m-%dT%H:%M:%SZ'), False)
+        batch_status = self._poll_on_pk_chunked_batch_status(job_id)
+        batch_status['job_id'] = job_id
+        # Close the job after all the batches are complete
+        self._close_job(job_id)
+
+        if batch_status['failed']:
+            LOGGER.info("Failed Bulk Query with window of date {} to {}".format(start_date_str, end_date.strftime('%Y-%m-%dT%H:%M:%SZ')))
+            # If batch_status is failed then reduce date window by half by updating end_date
+            # To updte end_date, substract half_day_range (i.e. half of the days between start_date and end_date)
```

Comment:
```suggestion
            # To update end_date, substract half_day_range (i.e. half of the days between start_date and end_date)
```

Reply:
We should probably note that in the event that `half_day_range` is an odd number, we will round down to the nearest integer because of the `//`

Reply:
Updated the comment and also added a comment related **half_day_range** odd number.